CHANGELOG.md

```diff
@@ -281,6 +281,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Deprecated function `pytorch_lightning.callbacks.device_stats_monitor.prefix_metric_keys` ([#11254](https://github.com/PyTorchLightning/pytorch-lightning/pull/11254))
 
 
+- Deprecated `on_configure_sharded_model` callback hook ([#11627](https://github.com/PyTorchLightning/pytorch-lightning/pull/11627))
```

Comment:
IMO it is important to mention which hook the user should migrate to, here in the changlog and in the deprecation warning message as well.